trackers/core/ksptracker/tracker.py

```diff
@@ -0,0 +1,252 @@
+
+                G.add_node(node)
+                node_dict[frame_idx].append(node)
+                self.node_to_detection[node] = (frame_idx, det_idx)
+
+                if frame_idx == 0:
+                    G.add_edge("source", node, weight=max(-node.confidence, 0.001))
+                if frame_idx == len(all_detections) - 1:
+                    G.add_edge(node, "sink", weight=0)
+
+        for i in range(len(all_detections) - 1):
+            for node in node_dict[i]:
+                for node_next in node_dict[i + 1]:
+                    dist = np.linalg.norm(
+                        np.array(node.position) - np.array(node_next.position)
+                    )
+                    if dist <= 2:
+                        G.add_edge(
+                            node_next,
+                            node,
+                            weight=max(dist - node_next.confidence, 0.001),
+                        )
+
+        return G
+
+    def _update_detections_with_tracks(
+        self, assignments: List[List[TrackNode]]
+    ) -> List[sv.Detections]:
+        """
+        Assign track IDs to detections.
+
+        Args:
+            assignments (Dict): Paths from KSP with track IDs
+
+        Returns:
+            List[sv.Detections]: Merged detections with tracker IDs
+        """
+        all_detections = []
+
+        assigned = set()
+        assignment_map = {}
+        for track_id, path in enumerate(assignments, start=1):
+            for node in path:
+                det_key = self.node_to_detection.get(node)
+                if det_key and det_key not in assigned:
+                    assignment_map[det_key] = track_id
+                    assigned.add(det_key)
+
+        for frame_idx, dets in enumerate(self.detection_buffer):
+            frame_tracker_ids = [-1] * len(dets)
+            for det_idx in range(len(dets)):
+                key = (frame_idx, det_idx)
+                if key in assignment_map:
+                    frame_tracker_ids[det_idx] = assignment_map[key]
+
+            dets.tracker_id = np.array(frame_tracker_ids)
+            all_detections.append(dets)
+
+        return all_detections
+
+    def ksp(self, graph: nx.DiGraph) -> List[List[TrackNode]]:
+        """
+        Find multiple disjoint shortest paths.
+
+        Args:
+            graph (nx.DiGraph): Detection graph
+
+        Returns:
+            List[List[TrackNode]]: Disjoint detection paths
+        """
+        paths: List[List[TrackNode]] = []
+        G_copy = graph.copy()
+
+        while True:
+            try:
+                path = nx.shortest_path(
+                    G_copy, source="source", target="sink", weight="weight"
+                )
+                if len(path) < 2:
+                    break
+                paths.append(path[1:-1])
+                for node in path[1:-1]:
+                    G_copy.remove_node(node)
+            except nx.NetworkXNoPath:
+                break
+
+        return paths
```

Comment:
This version of KSP doesn't seem to accurately match the algorithm described in Appendix B of the paper. I'm not entirely sure if this greedy search is guaranteed to find the globally optimal set of `k` paths.